addons/stock/models/stock_picking.py

```diff
@@ -83,8 +83,10 @@ def _default_show_operations(self):
 
     @api.model
     def create(self, vals):
-        if 'sequence_id' not in vals or not vals['sequence_id']:
-            if vals['warehouse_id']:
+        if 'sequence_id' not in vals or not vals.get('sequence_id'):
+            vals['sequence_code'] = vals.get('sequence_code', 'IN')
+            vals['code'] = vals.get('code', 'incoming')
+            if vals.get('warehouse_id'):
```

Comment:
We should not force the sequence code value
```suggestion
        if not vals.get('sequence_id') and vals.get('sequence_code'):
            if vals.get('warehouse_id'):
```
And, anyway, if the user does not provide neither the `sequence_id` nor the `sequence_code`, he will get a legit error on record creation since the second field is required
https://github.com/odoo/odoo/blob/4a4ae31882db94080809c600209453119c30f2b4/addons/stock/models/stock_picking.py#L39

Reply:
Hello @adwid 

Changes are done.
Thanks!